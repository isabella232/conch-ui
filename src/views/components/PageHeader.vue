<template>
    <section class="hero is-primary welcome is-small is-bold">
        <div class="hero-body">
            <h1 class="title">{{ title }}</h1>
            <h2 class="subtitle" v-if="subtitle">{{ subtitle }}</h2>
        </div>
    </section>
</template>

<script>
export default {
    props: {
        title: {
            type: String,
            required: true,
        },
        subtitle: {
            type: String,
            required: false,
            default: '',
        },
    },
};
</script>
