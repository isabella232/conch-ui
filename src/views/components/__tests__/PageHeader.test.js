import PageHeader from '../PageHeader.vue';
import { shallowMount } from '@vue/test-utils';

describe('PageHeader.vue', () => {
    let propsData;
    let subtitle;
    let title;
    let wrapper;

    beforeEach(() => {
        subtitle = 'Update password and profile settings';
        title = 'User Profile';
        propsData = { title: title, subtitle: subtitle };
        wrapper = shallowMount(PageHeader, { propsData });
    });

    test('should render correctly on initial render', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('should add a title from prop', () => {
        expect(wrapper.find('.title').html()).toContain(title);
    });

    test('should add a subtitle from prop', () => {
        expect(wrapper.find('.subtitle').html()).toContain(subtitle);
    });
});
