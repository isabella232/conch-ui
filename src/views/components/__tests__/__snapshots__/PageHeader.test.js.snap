// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader.vue should render correctly on initial render 1`] = `
<section class="hero is-primary welcome is-small is-bold">
  <div class="hero-body">
    <h1 class="title">User Profile</h1>
    <h2 class="subtitle">Update password and profile settings</h2>
  </div>
</section>
`;
